lib/parser.js

```diff
@@ -133,7 +133,13 @@ function parseFromUrl(url, fetchOptions, options) {
       .then(res => res.text())
       .then(doc => parse(doc, options))
       .then(result => resolve(result))
-      .catch(reject);
+      .catch(e => {
+        if (e instanceof ParserError) return reject(e);
+        return reject(new ParserError({
+          type: 'unexpected-error',
```

Comment:
shouldn't we maybe introduce a custom type here, like maybe `url-fetching-error` or `fetching-error`?
if you agree, don't forget about updating the readme

Reply:
Agree. `fetch-url-error` sounds good. Also, thanks for reminding to update the readme.

Reply:
awesome, so go ahead with the change, in the test also validate not only message but type too please